### PR TITLE
Blacklisted Baton and Bladed Caps from Pneumatic Cannon

### PR DESCRIPTION
**Pull request by FungiFellow**
## About the PR

I added the  `- CannonRestrict` Tag to Batons and Caps, because people were getting a bit too silly with both
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Fixed Batons and Flatcaps fitting in Pneumatic Cannon

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml [hunk 1]
```diff
@@ -167,6 +170,9 @@
     sprite: Clothing/Head/Hats/brownflatcap.rsi
   - type: Clothing
     sprite: Clothing/Head/Hats/brownflatcap.rsi
+  - type: Tag
+    tags:
+    - CannonRestrict
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml:175]: this gets inherited from `parent: BladedFlatcapGrey`
```suggestion
```

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml (hunk 1 above)
Comment (FungiFellow) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml]: Done.

**Comment by SoulFN** [comment]
Why everyone wants to make pneumatic cannon useless?

**Comment by superjj18** [comment]
> Why everyone wants to make pneumatic cannon useless?

Make it so emagging the pneumatic cannon removes the whitelist :trollface:

**Comment by SoulFN** [comment]
Pneumatic cannon is just better throwing, you can throw batons and flatcaps very fast without cannon, i think pneumatic cannon don`t need debuffs otherwise it will become useless, it also has a huge size which is a big disadvantage

**Comment by metalgearsloth** [comment]
Just turn it into a whitelist because the blacklist is clearly not working out if we have to keep making PRs to fix abuse.

**Comment by FungiFellow** [comment]
> Just turn it into a whitelist because the blacklist is clearly not working out if we have to keep making PRs to fix abuse.

It's easier to pick out what I don't want used in it than it is to select everything I want in it.

**Comment by potato1234x** [comment]
> Just turn it into a whitelist because the blacklist is clearly not working out if we have to keep making PRs to fix abuse.

How you gonna whitelist every basic item in the game? Blacklist is a lot easier here

**Comment by K-Dynamic** [comment]
Just make pneumatic cannon storage from 4x4 down to 2x4 or 3x4 :trollface:

**Comment by metalgearsloth** [comment]
> > Just turn it into a whitelist because the blacklist is clearly not working out if we have to keep making PRs to fix abuse.
> 
> How you gonna whitelist every basic item in the game? Blacklist is a lot easier here

1. You shouldn't whitelist every item in the game.
2. We keep having to make more and more PRs just to maintain this as the game grows.

**Comment by superjj18** [comment]
> Just make pneumatic cannon storage from 4x4 down to 2x4 or 3x4 :trollface:

Honestly yeah, 4x4 is gigantic for the equivalent of a potato gun.  I’d almost rather make it 2x2 and remove the whitelist

**Comment by SoulFN** [comment]
> Honestly yeah, 4x4 is gigantic for the equivalent of a potato gun. I’d almost rather make it 2x2 and remove the whitelist

Maybe just drop fire speed to 1 per second?

**Comment by superjj18** [comment]
> 
> 
> > Honestly yeah, 4x4 is gigantic for the equivalent of a potato gun. I’d almost rather make it 2x2 and remove the whitelist
> 
> 
> 
> Maybe just drop fire speed to 1 per second?
> 
> 

Why does a potato gun have an internal magazine?

**Comment by K-Dynamic** [comment]
> Why does a potato gun have an internal magazine?

It's like whoever made the launcher wanted it to be spammy

For real, it should have decreased storage and/or decreased firerate and/or no automatic mode

**Comment by QuietlyWhisper** [comment]
> Just turn it into a whitelist because the blacklist is clearly not working out if we have to keep making PRs to fix abuse.

Blacklisting items with throw damage might be the better route if this is what the problem is
Else I think someone else in the maint channel mentioned nerfing the fire rate or something with a cooldown

**Comment by LankLTE** [comment]
Throwing has a 0.5 second cooldown, so the pneumatic cannon having a similar (maybe slightly longer) cooldown should mean anything “broken” inside of it is just as broken if thrown and should be fixed separately. Least that’s my take on it.

**Comment by IProduceWidgets** [comment]
Its just a bug where thrown weapons hit twice for some reason. This is fixing the wrong breaky.

**Comment by superjj18** [comment]
> Its just a bug where thrown weapons hit twice for some reason. This is fixing the wrong breaky.

I seriously believe they should be twice as damaging since it’s a freakin cannon, the problem is it should be a breach-loading weapon that only fires one medium object a couple small objects all at once.

it shouldn’t be “load up 4 rounds of truncheon ammo” it should be “load a single truncheon so that it’s more effective than throwing it, and I have to be robust and reload it after every shot”

**Comment by EmoGarbage404** [comment]
superseded by #26878

**Pull request closed without merging**